tests/deployment_utils.py

```diff
@@ -178,24 +180,27 @@ def local_sources(tmpdir):
 
 
 class UniversumRunner:
-    def __init__(self, perforce_workspace, git_client, local_sources, nonci):
-        self.perforce = perforce_workspace
-        self.git = git_client
-        self.local = local_sources
-        self.nonci = nonci
+    def __init__(self, perforce_workspace: PerforceWorkspace,
+                 git_client: GitClient,
+                 local_sources: LocalSources,
+                 nonci: bool):
+        self.perforce: PerforceWorkspace = perforce_workspace
+        self.git: GitClient = git_client
+        self.local: LocalSources = local_sources
+        self.nonci: bool = nonci
 
         # Need to be initialized in 'set_environment'
-        self.environment = None
-        self.working_dir: str = None
-        self.project_root = None
-        self.artifact_dir = None
+        self.environment: ExecutionEnvironment = None  # type: ignore
```

Comment:
Why "type: ignore" instead of Optional?

Reply:
Because otherwise Mypy generates lots of warnings later that are basically 'your object might be None and you cannot treat it like ExecutionEnvironment/py.path.local/etc.'
And these warnings would be correct if we didn't manually make sure to set all this objects to `not None` before calling any function that may refer to them.